app/instrumentation-tests/src/org/commcare/androidTests/LazyVideosTests.kt

```diff
@@ -0,0 +1,152 @@
+package org.commcare.androidTests
+
+
+import android.content.Intent
+import android.os.Build
+import android.util.Log
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.RootMatchers
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.filters.SdkSuppress
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.commcare.CommCareApplication
+import org.commcare.annotations.BrowserstackTests
+import org.commcare.dalvik.R
+import org.commcare.utils.CustomMatchers
+import org.commcare.utils.InstrumentationUtility
+import org.commcare.utils.isPresent
+import org.hamcrest.CoreMatchers
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.allOf
+import org.junit.After
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+class LazyVideosTests : BaseTest() {
+
+    companion object {
+        const val CCZ_NAME = "lazy_videos_tests.ccz"
+        const val APP_NAME = "Lazy Videos"
+    }
+
+    @Before
+    fun setup() {
+        InstrumentationUtility.uninstallCurrentApp()
+        InstrumentationUtility.installApp(CCZ_NAME)
+        Espresso.pressBack()
+        updateApp()
+
+    }
+
+    @After
+    fun tearDown() {
+        InstrumentationUtility.logout()
+    }
+
+    @Test
+    fun testVideosWithReferences() {
+        testVideosWithValidReference()
+        testVideosWithNoReferences()
+    }
+
+    private fun updateApp(){
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("Go To App Manager"))
+            .perform(click())
+        onView(withText(APP_NAME)).perform(click())
+        onView(withText("Update"))
+            .perform(click())
+        InstrumentationUtility.waitForView(withSubstring("Update to version"))
+        gotoLogin()
+    }
+
+    private fun gotoLogin() {
+        for (i in 0..2) { // Try atmost 3 times.
+            if (onView(withId(R.id.edit_username)).isPresent()) {
+                return
+            } else {
+                Espresso.pressBack()
+            }
+        }
+    }
+    private fun testVideosWithValidReference() {
+        InstrumentationUtility.login("test1", "123")
+        InstrumentationUtility.openForm(1, 0)
+        InstrumentationUtility.waitForView(withId(R.id.video_button))
+        if ((onView(withTagValue(CoreMatchers.`is`(R.drawable.update_download_icon))).isPresent())
```

Comment:
this should always be present now right ?

Reply:
yes.

Reply:
we should not put this in `if` condition then

Reply:
But if we keep it, then even if the .ccz is updated then also this will work. Thats what I thought of. If you think it should be removed, I can do that.

Reply:
We should remove it as we don't want to paas this test unless it goes through the download workflow.

Reply:
removed